src/pages/RegisterView.jsx

```diff
@@ -53,30 +53,55 @@ function RegisterView() {
     const passwordResponse = validatePassword(password);
     const reenteredPasswordResponse = validateReenteredPassword(password, reenteredPassword);
 
-    if (emailResponse !== SUCCESS) {
-      setEmailErrorMessage(emailResponse);
-    }
-    else {
-      setEmailErrorMessage(null);
-    }
+    handleEmailValidationResponse(emailResponse);
+    handlePasswordValidationResponse(passwordResponse);
+    handleReenteredPasswordValidationResponse(reenteredPasswordResponse);
 
-    if (passwordResponse !== SUCCESS) {
-      setPasswordErrorMessage(passwordResponse);
-    }
-    else {
-      setPasswordErrorMessage(null);
-    }
+    return (emailResponse === SUCCESS) && (passwordResponse === SUCCESS) && (reenteredPasswordResponse === SUCCESS);
+  }
 
-    if (reenteredPasswordResponse !== SUCCESS) {
-      setReenteredPasswordErrorMessage(reenteredPasswordResponse);
-    }
-    else {
-      setReenteredPasswordErrorMessage(null);
-    }
+  function emailValidationCheck(email) {
+    setEmail(email);
+    handleEmailValidationResponse(validateEmail(email));
```

Comment:
Why a separate function to handle the response? Logic might be easier to follow if you merge the  two functions.

```javascript
setEmail(email);

const response = validateEmail(email);
setEmailErrorMessage(response === 'SUCCESS' ? null : response);
```
I guess that also means you'd have to rename the original function since it's both handling the change event and validating the input.

Reply:
Usually i like to avoid duplicate logic and keeping functions broken out and simplified, since these are pretty specific to setting the state so i can put that back.